actions/loadAddDeck.js

```diff
@@ -10,7 +11,32 @@ export default function loadAddDeck(context, payload, done) {
       && (store.userid !== undefined && store.userid !== null && store.userid !== '')
       && (store.jwt !== undefined && store.jwt !== null && store.jwt !== ''))) {
         context.executeAction(navigateAction, {url: '/'});//, reqId: payload.navigate.reqId});
-        $('.ui.login.modal').modal('show');
+        //prepraring the modal
+        context.executeAction(updateTrap,{activeTrap:true});
+        //hidden the other page elements to readers
+        $('#app').attr('aria-hidden','true');
+
+        $('.ui.login.modal')
+        .modal({
+            onVisible: () => {
```

Comment:
Why is the configuration needed here again? Header.js should only configure the modal and the other places which opens the modal should just open it.
Thus the following should be enough:
context.executeAction(updateTrap,{activeTrap:true});
 $('#app').attr('aria-hidden','true');
$('.ui.login.modal'). modal('show');

Reply:
I think that the semantic ui has some problem when is called from an action, and not from a component. When the modal is open from a compoenent, you can see how the class of the body tag changes to dimmable dimmed. This not happens when it is shown from the action, and the dimmer is not hidden. If we add this class using jquery (as automatically occurs when you show the modal from a compoennt), when you close the modal, the dimmer is hidden. 
For this reason, I have commented this onVisible setting:
//This class is added automatically when the modal is called from a component, not an action.
 //If it is not set, the dimmer is not disabled when the modal is hidden

Reply:
OK the reason should be that we have an isomorphic application (sometimes actions are executed on the server which have to initiate the modal again)
But if you configure the modal in Header.js you dont need to do that again in the other components, right?

Reply:
I think you configure it each time you shows it...and in header is not needed to configure the onVisible...because it works ok.

Reply:
and...in header the modal is only configured if ou clicks on SigIn button.

All these problems disappear if the nodal from react-semantic ui is used, because you control the modal using the same way than other components. But, I tried to patch this component using the current code, and transform it to a totally react comeponent, as it has several code....

Reply:
I created a branch SWIK-2065-fixSignInModal_proposal in where the LoginModal is initiated just one time in the Header. Please check if this is interesting to be merged.

In every case please check that LoginModal uses itself in the preferred way. At the moment there are many calls in there which just do: $('.ui.login.modal').modal('toggle');

Reply:
I think this hould be checked for the original programmer who choose use toggle instead...I have tried to undertand all the code, and I think than toggle was use to hide in all cases...but original programmers should check it.

Reply:
@TBoonX  is going to finish it, now he has the ticket assigned